app/controllers/SubmitPoaController.scala

```diff
@@ -61,7 +70,147 @@ class SubmitPoaController @Inject()(cc: MessagesControllerComponents,
     }
   }
 
-  def extractNino(json: JsValue): Option[String] = {
+  private def overwriteTotalAmount(nino: String, json: JsValue): Unit = {
+    (extractPoAAmount(json), extractTaxYear(json)) match {
```

Comment:
if we can refactor this method pls? you can move nested pattern matching into private methods etc